examples/directml/llm/llm.py

```diff
@@ -52,7 +52,7 @@ def set_config_parameters(tokenizer: transformers.AutoTokenizer, repo_id: str, n
 
     config.hidden_size = llm_model.config.hidden_size
     config.num_heads = llm_model.config.num_attention_heads
-    config.head_dim = getattr(llm_model.config, "head_dim", config.hidden_size / config.num_heads)
+    config.head_dim = getattr(llm_model.config, "head_dim", config.hidden_size // config.num_heads)
```

Comment:
thanks for the change. did you hit this error when testing with a model?

Reply:
Yes, I'm getting the following error trace. This is because the config.head_dim is of type `float` and it is expected to be `int`

```sh
decoder_model.py 244 __init__
self.q_proj = torch.nn.Linear(config.hidden_size, config.num_heads * config.head_dim, bias=config.use_bias)

linear.py 98 __init__
self.weight = Parameter(torch.empty((out_features, in_features), **factory_kwargs))

TypeError:
empty() received an invalid combination of arguments - got (tuple, dtype=NoneType, device=NoneType), but expected one of:
 * (tuple of ints size, *, tuple of names names, torch.memory_format memory_format, torch.dtype dtype, torch.layout layout, torch.device device, bool pin_memory, bool requires_grad)
 * (tuple of ints size, *, torch.memory_format memory_format, Tensor out, torch.dtype dtype, torch.layout layout, torch.device device, bool pin_memory, bool requires_grad)
 ```

Reply:
Thanks! that makes sense. for gemma model, it's model config has `head_dim` attribute but it's not there for phi3, etc.